cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -834,6 +1355,10 @@ class TableRadixSorter {
 // Sort a table using a single sort and multiple-key comparisons.
 class MultipleKeyTableSorter : public TypeVisitor {
  private:
+  // TODO instead of resolving chunks for each column independently, we could
+  // split the table into RecordBatches and pay the cost of chunked indexing
+  // at the first column only.
```

Comment:
Can we always do it?
My understanding that each chunked array in a table can have different number of chunks. For example, the table is valid:

```text
a: [[0, 1], [2, 3, 4]]
b: [[10], [11, 12], [13], [14]]
```

I'm not sure we can split the table into record batches efficiently.

Reply:
`TableBatchReader` can be used for that.